docs/source/developers/java/building.rst

```diff
@@ -0,0 +1,308 @@
+Go to parent project and execute:
+
+.. code-block::
+
+    $ cd arrow/java
+    $ export JAVA_HOME=<absolute path to your java home>
+    $ java --version
+    $ mvn clean install
+    [INFO] Reactor Summary for Apache Arrow Java Root POM 8.0.0-SNAPSHOT:
+    [INFO]
+    [INFO] Apache Arrow Java Root POM ......................... SUCCESS [  4.922 s]
+    [INFO] Arrow Format ....................................... SUCCESS [  4.395 s]
+    [INFO] Arrow Memory ....................................... SUCCESS [  0.938 s]
+    [INFO] Arrow Memory - Core ................................ SUCCESS [ 11.491 s]
+    [INFO] Arrow Memory - Unsafe .............................. SUCCESS [  5.449 s]
+    [INFO] Arrow Memory - Netty ............................... SUCCESS [  8.754 s]
+    [INFO] Arrow Vectors ...................................... SUCCESS [01:37 min]
+    [INFO] Arrow Compression .................................. SUCCESS [  5.907 s]
+    [INFO] Arrow Tools ........................................ SUCCESS [ 11.015 s]
+    [INFO] Arrow JDBC Adapter ................................. SUCCESS [ 13.751 s]
+    [INFO] Arrow Plasma Client ................................ SUCCESS [  3.934 s]
+    [INFO] Arrow Flight ....................................... SUCCESS [  0.649 s]
+    [INFO] Arrow Flight Core .................................. SUCCESS [ 38.331 s]
+    [INFO] Arrow Flight GRPC .................................. SUCCESS [  5.894 s]
+    [INFO] Arrow Flight SQL ................................... SUCCESS [ 21.862 s]
+    [INFO] Arrow Flight Integration Tests ..................... SUCCESS [  7.638 s]
+    [INFO] Arrow AVRO Adapter ................................. SUCCESS [ 14.447 s]
+    [INFO] Arrow Algorithms ................................... SUCCESS [ 35.105 s]
+    [INFO] Arrow Performance Benchmarks ....................... SUCCESS [  5.323 s]
+    [INFO] ------------------------------------------------------------------------
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------\
+
+Advanced Installation
+---------------------
+
+Build C / C++ Libs
+++++++++++++++++++
+
+Build C / C++ Libs Ubuntu
+#########################
+
+For Build C Data Interface lib & Build C++ Libs. Consider: Next step requires: Docker & Docker Compose.
+
+.. code-block::
+
+    $ cd arrow
+    $ source dev/conbench_envs/hooks.sh install_archery
+    Successfully installed archery
+    $ archery docker run java-jni-manylinux-2014
+    $ ls -latr java-dist/
+    |__ libarrow_cdata_jni.so
+    |__ libarrow_dataset_jni.so
+    |__ libarrow_orc_jni.so
+    |__ libgandiva_jni.so
+
+Build C / C++ Libs MacOS
+########################
+
+For Build C Data Interface lib:
+
+.. code-block::
+
+    $ cd arrow
+    $ brew bundle --file=cpp/Brewfile
+    Homebrew Bundle complete! 25 Brewfile dependencies now installed.
+    $ export GITHUB_WORKSPACE=<absolute path to your arrow project>
```

Comment:
Let's not use CI-specific variables, we can just get rid of the env var and use `./java_dist`, etc

Reply:
Changed